### DISCUSSION TITLE
blog/fully-reusable-components

### DISCUSSION THREAD

**Discussion by giscus** [theodorusclarence/theodorusclarence.com#391] [General]
# blog/fully-reusable-components

Creating a component is fairly easy, but doing them correctly is a different story.

https://theodorusclarence.com/blog/fully-reusable-components

**Comment by theodorusclarence** [comment]
Hello, kindly leave a like (or 5 😄), and comment if you got something to ask! 🚀

**Reply by idhamrsd** [reply]
So useful, thanks for writing this, will wait for the next one.

**Comment by yaffalhakim1** [comment]
Nice article, and useful even for me as beginner :)
one question, do you have repo for the example? so i can see the whole code and can understand them. thank you

**Reply by theodorusclarence** [reply]
Yes, you can check this https://github.com/theodorusclarence/ts-nextjs-tailwind-starter/blob/main/src/components/buttons/Button.tsx

It's an example of a more complex usage of fully reusable component

**Comment by krowter** [comment]
Nice summary!

Small nitpick: better make it explicit that children comes from product.title; not from Card's own children
`Omit<React.ComponentPropsWithoutRef<'div'>, 'children'>`
just to prevent this :D
`<Card product={{ title: 'some content' }}> some other content </Card>`

**Reply by theodorusclarence** [reply]
Totally agreed!

**Comment by Milkywayrules** [comment]
🙌🙌🙌

**Comment by zaaakher** [comment]
Great read with helpful lessons 👍

+1 for teaching us about ref forwarding in your next article (although I heard forwardRef is going to be deprecated and there's possibly a better alternative)
